Connecting a Frontend to a Backend Tutorial.

jeffmendoza opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1996)
<!-- Reviewable:end -->

steveperry-53 commented on docs/tutorials/connecting-apps/connecting-apps.md:
```diff
@@ -0,0 +1,162 @@
+
+{% capture prerequisites %}
+
+* {% include task-tutorial-prereqs.md %}
```
I think the prerequisites need to include using a cloud provider, like what I've done in https://github.com/kubernetes/kubernetes.github.io/pull/2006.

jeffmendoza replied:
Same comment here as there, I hope we can have tutorials with `type: LoadBalancer` not require a cloud provider.

jaredbhatti commented:
Ping, what's the status of this?

jeffmendoza commented:
Ah, looks like were were deciding on `type: LoadBalancer` and if we need to list a cloud provider as a prereq. Minikube does work with `type: LoadBalancer`, but we probably want a page to link to

chenopis commented:
@jeffmendoza @steveperry-53 Will that be addressed in this PR?

jeffmendoza commented:
@steveperry-53 Rebased and fixed merge conflicts. Updated prereqs, but as we discussed we will need more work in this area for all tutorials.

@chenopis no, we still need more detailed explanation of `type: LoadBalancer` in relation to Tutorials, which environments support it, and how to use `type: NodePort` if your environment does not.

steveperry-53 commented:
@jeffmendoza, What's the next step for me? Are you ready for me to take a final look and then merge?

jeffmendoza commented:
@steveperry-53 Yes, please.

steveperry-53 commented:
@jeffmendoza, See https://github.com/kubernetes/kubernetes.github.io/pull/1996/commits/1fdfd944176cfae05c382f356fa9e5cef149e19e for suggested edits.

Most of the edits are about these areas in our [style guide](https://kubernetes.io/docs/contribute/style-guide/): 1) present tense 2) not using "we" 3) code snippet format.

A few questions/comments:

1.  We should probably change the filename from connecting-apps.md to something like connect-frontend-backend.md.

2. The word "service" is overloaded in this topic. Could we use the word "application" unless we're talking about a Kubernetes Service? I like the way you've phrased this objective: "Use a Service object to connect the frontend application to the backend application."

3. Would it make sense to shorten the configuration files by removing fields that are incidental in this tutorial: resourceLimits, livenessProbe, etc?

jeffmendoza commented:
Thanks for the fixes, they look great.

1. SG
2. Sounds good too, I'll do a pass, but I might need your help.
3. Yes, I'll trim those  (this config is from a more complete example).

jeffmendoza commented:
@steveperry-53 updated.

steveperry-53 commented:
@jeffmendoza, If it were me, I'd eliminate the word "microservice" too. 

This tutorial shows how to connect a frontend application to a backend application by using a Kubernetes Service object.

* Use a Deployment to create and run a a backend application.
* Use a Deployment to create and run a frontend application.
* Use a Service object to connect the frontend to the backend.

I'll leave it up to you. I think we could publish this today. Would you like to make your final changes and squash?

jeffmendoza commented:
Removing "microservice" is a little problematic as that is the point of having the Nginx frontend, theoretically it would reverse proxy based on URL to many "microservices". Generically we are connecting applications, that is the goal of the tutorial, but the example is a "microservice" setup.

steveperry-53 commented:
SGTM

jeffmendoza commented:
@steveperry-53 squashed

jeffmendoza commented:
Thanks Steve
